drakrun/drakrun/test/test_vm.py

```diff
@@ -0,0 +1,257 @@
+import pytest
+
+from drakrun.config import InstallInfo
+from drakrun.vm import VirtualMachine
+from drakrun.util import safe_delete
+from _pytest.monkeypatch import MonkeyPatch
+from common_utils import remove_files, tool_exists
+import tempfile
+import subprocess
+import os
+import re
+import logging
+
+
+@pytest.fixture(scope="session")
+def monkeysession(request):
+    mp = MonkeyPatch()
+    yield mp
+    mp.undo()
+
+
+@pytest.fixture(scope="module")
+def patch(monkeysession):
+    if not tool_exists('xl'):
+        pytest.skip("xen is not found")
+
+    def install_patch():
+        return InstallInfo(
+            vcpus=1,
+            memory=512,
+            storage_backend='qcow2',
+            disk_size='200M',
+            iso_path=None,  # not being required
+            zfs_tank_name=None,
+            lvm_volume_group=None,
+            enable_unattended=None,
+            iso_sha256=None
+        )
+    monkeysession.setattr(InstallInfo, "load", install_patch)
+    monkeysession.setattr(InstallInfo, "try_load", install_patch)
+
+    # being yielded so the the monkeypatch doesn't start cleanup if returned
+    yield monkeysession
+
+
+@pytest.fixture(scope="module")
+def test_vm(patch):
+    monkeysession = patch
+
+    @property
+    def vm_name(self):
+        return "test-hvm64-example"
+    monkeysession.setattr(VirtualMachine, "vm_name", vm_name)
+    test_vm = VirtualMachine(None, 0, "test-hvm64-example")
+
+    yield test_vm
+
+
+@pytest.fixture(scope="module")
+def config():
+    tmpf = tempfile.NamedTemporaryFile(delete=False).name
+    module_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..')
+    cfg_path = os.path.join(module_dir, "tools", "test-hvm64-example.cfg")
+    firmware_path = os.path.join(module_dir, "tools", "test-hvm64-example")
+
+    with open(cfg_path, 'r') as f:
+        test_cfg = f.read().replace('{{ FIRMWARE_PATH }}', firmware_path).encode('utf-8')
+
+    with open(tmpf, 'wb') as f:
+        f.write(test_cfg)
+
+    yield tmpf
+    safe_delete(tmpf)
```

Comment:
Why not use `with tempfile.NamedTemporaryFile` instead of `delete=False` and safe_delete?

Reply:
I don't really know actually tbh, I had a lot of trouble setting this temporary files thing up which can stay active throughout the module. I had also tried `tmp_path` and `tmp_path_factory` given by pytests but they also created a lot of issues for me, So i just stuck to whatever worked for me during `lvm_tests` and picked the same format

Reply:
Ok :thinking: